File: Easy/AddBinary.cpp
#include <iostream>
#include<math.h>
using namespace std;

string addBinary(string a, string b) {
    string result = "";
    int carry = 0;
    int i = a.length() - 1;
    int j = b.length() - 1;
    int x,y;
    while (i >= 0 || j >= 0 || carry) {
        int x = i >= 0 ? a[i] - '0' : 0;
        int y = j >= 0 ? b[j] - '0' : 0;
        int sum = x + y + carry;
        result = to_string(sum % 2) + result;
        carry = sum / 2;
        i--;
        j--;
    }

    return result;

}

int main(){
    cout << addBinary("11", "10");
}

File: Easy/AddBinary.py
def addBinary(a: str, b: str) -> str:
    # num_a = int(a, 2)
    # num_b = int(b, 2)
    # sum = num_a + num_b
    # res = bin(sum)[2:]
    return str(bin(int(a, 2) + int(b, 2))[2:])
    
    

print(addBinary("11", "1"))